conda-store-server/conda_store_server/app.py

```diff
@@ -471,6 +474,46 @@ def celery_app(self):
         self._celery_app.config_from_object(self.celery_config)
         return self._celery_app
 
+    @property
+    def plugin_registry(self):
+        if hasattr(self, "_plugin_registry"):
+            return self._plugin_registry
+
+        self._plugin_registry = plugin_registry.PluginRegistry()
+        self._plugin_registry.collect_plugins()
+        return self._plugin_registry
+
+    @property
+    def plugin_manager(self):
+        if hasattr(self, "_plugin_manager"):
+            return self._plugin_manager
+
+        self._plugin_manager = pluggy.PluginManager(hookspec.spec_name)
+        self._plugin_manager.add_hookspecs(hookspec.CondaStoreSpecs)
+
+        # Get settings - required to configure the lock plugin with the
+        # correct conda command
+        with self.session_factory() as db:
+            settings = self.get_settings(db)
+
+        # Load lock plugin
+        self.load_plugin_by_name(
```

Comment:
currently loading the lock plugin globally by it's name. In the future, when more lock plugins are available, this should be configurable. See the[ pluggy demo  ](https://github.com/conda-incubator/conda-store/compare/main...soapy1:conda-store:plugin-demo-pluggy#diff-718d9a91e39401c644e5ff6dc07145f1530a2b13204a17d0b8a4a1d37174b84dR34) for more examples of how this can be done.